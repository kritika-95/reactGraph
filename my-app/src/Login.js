// app/src/Login.js
import React from 'react';
class Login extends React.Component {
    constructor(props) {
  super(props);
  this.state = {
   username: '',
   password: '',
   date:''
  };
 
 }
 handleInputChange = (event) => {
  this.setState({
   [event.target.name]: event.target.value
  })
};


 render() {
  return (
   <form className='login'>
     <label>Username</label>
     <input id='email123' onChange={this.handleInputChange} name='email' type='text' placeholder="mail.." /><br/><br/>
     <label>Password</label>
     <input id='password' onChange={this.handleInputChange} name='password' type='password' placeholder="password.." /><br/><br/>
     <label>Date</label>
     <input type="date" readOnly={false} required={true}/>
     <button id='01' disabled={true}  className="button">Submit</button>
   </form>
  )
 }
}
export default Login;