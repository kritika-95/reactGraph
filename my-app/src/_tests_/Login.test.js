// Login-test.js
import React from 'react';
import { shallow } from 'enzyme';
import Login from '../Login';
import renderer from 'react-test-renderer';

describe('login components',()=>{
// make our assertion and what we expect to happen 
 it('should render without throwing an error', () => {
   expect(shallow(<Login />).find('form.login').exists()).toBe(true)
 });

 // within the Login components describe function
 it('renders a email input', () => {
  expect(shallow(<Login />).find('#email123').length).toEqual(1)
 });

 it('renders a password input',()=>{

    expect(shallow(<Login/>).find('#password').length).toEqual(1)
 });
 it('renders status of button',()=>{
    expect(shallow(<Login/>).find('button').prop('disabled')).toBe(true);
 });
it('renders readonly property of date',()=>{
    expect(shallow(<Login/>).find('button').prop('readOnly')).toBeFalsy();
});
it('renders required property of input',()=>{
    expect(shallow(<Login/>).find('input').prop('required')).toBe(true);
});
});




// within the Login components describe function
describe('Email input', () => {
  
  it('should respond to change event and change the state of the Login Component', () => {
   
   const wrapper = shallow(<Login />);
   wrapper.find('#email123').simulate('change', {target: {name: 'email', value: 'kritika@gmail.com'}});
   
  expect(wrapper.state('email')).toEqual('kritika@gmail.com');
  })
 });

 describe('incorrect format for email', () => {
  
  it('should respond to change event and change the state of the Login Component', () => {
   
   const wrapper = shallow(<Login />);
   wrapper.find('#email123').simulate('change', {target: {name: 'value', value: 'kritika'}});
   
  expect(wrapper.state('email')).not.toEqual('kritika');
  })
 });



describe('Password input',()=>{
    it('should respond to change event and change the state of the login copmonent',()=>{
        const wrapper=shallow(<Login/>);
        wrapper.find('#password').simulate('change',{target:{name:'password', value:'pwd'}});

        expect(wrapper.state('password')).toEqual('pwd');
    })
});
describe('placeholder for username',()=>{
    it('should render correct placeholder for mail',()=>{
        const wrapper=shallow(<Login/>);
        wrapper.find('#email123').simulate('change',{target:{name:'placeholder',value:'mail..'}});
        expect(wrapper.state('placeholder')).toEqual('mail..');
    })
});

describe('placeholder for password',()=>{
    it('should render correct placeholder for password',()=>{
        const wrapper=shallow(<Login/>);
        expect(wrapper.find('#password').at(0).props().placeholder).toEqual('password..');
    })
});



//snapshot testing for login form components
 test('when login components will change',()=>{
  const component= renderer.create(
      <Login/>
  );
  let tree=component.toJSON();
 expect(tree).toMatchSnapshot();

 });

