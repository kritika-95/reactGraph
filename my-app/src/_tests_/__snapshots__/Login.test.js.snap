// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when login components will change 1`] = `
<form
  className="login"
>
  <label>
    Username
  </label>
  <input
    id="email123"
    name="email"
    onChange={[Function]}
    placeholder="mail.."
    type="text"
  />
  <br />
  <br />
  <label>
    Password
  </label>
  <input
    id="password"
    name="password"
    onChange={[Function]}
    placeholder="password.."
    type="password"
  />
  <br />
  <br />
  <label>
    Date
  </label>
  <input
    readOnly={false}
    required={true}
    type="date"
  />
  <button
    className="button"
    disabled={true}
    id="01"
  >
    Submit
  </button>
</form>
`;
